sklearn/linear_model/_bayes.py

```diff
@@ -32,9 +33,20 @@ class BayesianRidge(RegressorMixin, LinearModel):
 
     Parameters
     ----------
-    n_iter : int, default=300
+    n_iter : int
         Maximum number of iterations. Should be greater than or equal to 1.
 
+        .. deprecated:: 1.2
+           ``n_iter`` is deprecated in 1.2 and will be removed in 1.4. Use
+           ``max_iter`` instead.
+
+    max_iter : int, default=300
+        Maximum number of iterations over the complete dataset before
+        stopping independently of any early stopping criterion.
+        If ``max_iter`` is not None, ``n_iter`` is ignored.
+
+        .. versionadded:: 1.2
```

Comment:
Maybe it would be better to have a `versionchanged` mentioning that `n_iter` is renamed to `max_iter` in 1.3